pulser/pulse.py

```diff
@@ -76,12 +83,18 @@ def __init__(self, amplitude, detuning, phase, post_phase_shift=0):
                              "non-negative.")
         self.amplitude = amplitude
         self.detuning = detuning
+        phase = cast(float, phase)
         self.phase = float(phase) % (2 * np.pi)
+        post_phase_shift = cast(float, post_phase_shift)
         self.post_phase_shift = float(post_phase_shift) % (2 * np.pi)
 
     @classmethod
     @parametrize
-    def ConstantDetuning(cls, amplitude, detuning, phase, post_phase_shift=0):
+    def ConstantDetuning(cls, amplitude: Waveform,
```

Comment:
Don't forget the `Parametrized` option for the `Waveforms` on the classmethods.

Reply:
> Don't forget the `Parametrized` option for the `Waveforms` on the classmethods.

How will it work with this check in `__init__` ?

```
        if not (isinstance(amplitude, Waveform) and
                isinstance(detuning, Waveform)):
            raise TypeError("'amplitude' and 'detuning' have to be waveforms.")
```

Reply:
It's the same as with the `Waveform`s, when it's parametrized it doesn't run (in this case, because of the `@parametrize` decorator). You should cast to `Waveform` inside the function block so that `mypy` doesn't complain, though.